### PR TITLE
[JS] Generate in multiple files, added copyright support & fixed HEADER.inc path

### LINKED ISSUE #10: [JS] Output the generated code in multiple files
the code was initially output into a single generated file, as the project matures this should be done in multiple clear schema files with dependencies as needed

### PR DESCRIPTION
**Pull request by sleepyOwl14**
Fixed #10 and #6 
Also fixed the relative path for HEADER.inc

### CODE REVIEW

**Comment by Vektrat** [comment]
there's a hard check for the linter, this is why tests are not passing, make sure to be strict with it ^^

also, for the future, try making small concise commits, in this PR it doesn't matter much because thankfully they are short, but it'd be nice to have commits that have a specific target, and not a PR that fixes 2-3 different things (easier to track later on, because we "merge" with squash)

File: README.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ python main.py [OPTIONS]
 | -o, --output TEXT    | output directory                                        | _generated    |
 | -i, --include TEXT   | schema root directory                                   | ./schemas     |
 | -g, --generator TEXT | generator to use to produce output files                |               |
-| -c, --copyright TEXT | file containing copyright data to use with output files | ../HEADER.inc |
+| -c, --copyright TEXT | file containing copyright data to use with output files | ./HEADER.inc |
```
[the remark this answers is not in the document]
Comment (sleepyOwl14) [reply README.md]: that is the default copyright file path when copyright file parameter is not presented, using `../` is not making any sense, it is out of the project directory

**Comment by sleepyOwl14** [comment]
fixed issue #15

File: generators/javascript/JavaScriptGenerator.py [hunk 2]
```diff
@@ -93,18 +93,36 @@ def __init__(self, schema, options):
         self.serialize_method = None
         self.consumer_class = None
         self.exports = None
+        self.options = options
+        self.current = None
```
[the remark this answers is not in the document]
Comment (sleepyOwl14) [reply generators/javascript/JavaScriptGenerator.py]: changed variable name to `schema_items_iterator`

**Comment by Vektrat** [comment]
Please consider the comment I already put two times before, maybe I was not clear enough with "commits" - what I was suggesting was to create a PR for each different feature, fix, or enhancement that is done, otherwise this PR/branch will grow to be unmanageable

**Comment by sleepyOwl14** [comment]
don't worry, this will be my last commit, not gonna fix all the issues. I will just leave it here. 

- I think it is better to generate out the utils class, as almost all the buffer class is using it.
- I know it is not what you wanted for generate into multiple files, but it is important, at least it is usable. It can use `scripts/generate_all.sh` to generate all transactions class into their own respective file

Thanks

**Comment by sleepyOwl14** [comment]
I think I will close this PR, I will create a new PR

**Pull request closed without merging**